app/lib/dns.server.ts

```diff
@@ -157,8 +158,30 @@ export const getChangeStatus = async (changeId: string) => {
   }
 };
 
-const isNameValid = (name: string) => {
-  return /^[a-z0-9-]+.[a-z0-9-]+.[a-z]+.[a-z]+\.?$/.test(name);
+//Domain name rules
+//1. Domain name pattern should be [name].[studentId].rootDomain.com(.)
+//2. Domain name can contain only alphanumerical characters, '-', and '_'
+//3. Domain name should not start or end with -
+//4. Domain name cannot contain multiple consecutive '-' or '_'
+//5. Domain name can contain uppercase in UI but it is converted to lowercase before validation
+//6. studentId only contains alphanumerical characters
+export const isNameValid = (name: string) => {
+  const rootDomain: string = process.env.ROOT_DOMAIN!;
+  let subdomain: string;
+
+  //It only validates domain name and studentId, not root domain
+  if (name.endsWith('.')) {
+    subdomain = name.replace(`.${rootDomain}.`, '');
+  } else {
+    subdomain = name.replace(`.${rootDomain}`, '');
+  }
+  return (
+    /^(?!.*[-_]{2,})(?!^[-])[a-z0-9_-]+[^-_]\.[a-z0-9]+$/.test(subdomain) &&
```

Comment:
I would remove [studentId] from the subdomain too like:
```js
const postfix = `.${studentId}.${rootDomain}`;
if (!name.endsWith(postfix)) {return false;};

const subdomain = name.substring(0, name.length - postfix);
```

now you can validate only the [name] part, without having to worry about any dots
Very impressive regex work but 'aaa..bbb' still passes :( possible that it would be caught by isFQDN

Reply:
Thanks for checkout out! Yes, it would be much easier without `studentId`. I will have to speak with @sfrunza13 if I can easily get `studentId` info. I will add more test cases and improve it.

Reply:
We can't get student IDs.  We're only going to get their username (bit before `@myseneca.ca` or `@senecacollege.ca` for employees).  It may contain a `.`, letters, numbers, a `-`.

Reply:
I think we are talking about the same thing, it's just not named correctly ... What would be a good name for that? In an email that would be called `localpart` but I'm sure we are not going to call that

Reply:
As far as the SAML claim we get, it's the `SAML account name`, so maybe `account name` or `username` is good enough?